pkg/drawio/treeNodeInterface.go

```diff
@@ -86,62 +88,77 @@ func setGeometry(tn TreeNodeInterface) {
 }
 
 // /////////////////////////////////////////////////////////////////////
-// getAllNodes() - return all the nodes in the sub tree
-func getAllNodes(tn TreeNodeInterface) []TreeNodeInterface {
-	childrenSet := map[TreeNodeInterface]bool{}
-	squares, icons, lines := tn.children()
-	for _, s := range squares {
-		childrenSet[s] = true
-	}
-	for _, i := range icons {
-		childrenSet[i] = true
+// getSubTreeNodes() - return all the nodes in the sub tree
+
+func getSubTreeNodes(tn TreeNodeInterface) (
+	squares []SquareTreeNodeInterface,
+	icons []IconTreeNodeInterface,
+	lines []LineTreeNodeInterface) {
+	squares, icons, lines = tn.children()
+	for _, child := range joinTnsLists(squares, icons, lines) {
+		subSquares, subIcons, subLines := getSubTreeNodes(child)
+		squares = append(squares, subSquares...)
+		icons = append(icons, subIcons...)
+		lines = append(lines, subLines...)
 	}
-	for _, l := range lines {
-		childrenSet[l] = true
+	switch {
+	case tn.IsSquare():
+		squares = append(squares, tn.(SquareTreeNodeInterface))
+	case tn.IsIcon():
+		icons = append(icons, tn.(IconTreeNodeInterface))
+	case tn.IsLine():
+		lines = append(lines, tn.(LineTreeNodeInterface))
 	}
-	for child := range childrenSet {
-		sub := getAllNodes(child)
-		for _, s := range sub {
-			childrenSet[s] = true
-		}
-	}
-	childrenSet[tn] = true
-	ret := []TreeNodeInterface{}
-	for c := range childrenSet {
-		ret = append(ret, c)
-	}
-	return ret
+	// remove duplications:
```

Comment:
Why do we have duplicates?

Reply:
same icons can be in both a groupsquare, subnet, and/or security group

Reply:
fixed